cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureStorage.java

```diff
@@ -162,7 +162,7 @@ private String buildStorageName(ArmAttachedStorageOption armAttachedStorageOptio
             }
             result = name + storageType.getAbbreviation() + paddedId + new BigInteger(1, digest).toString(RADIX);
         } catch (NoSuchAlgorithmException ignored) {
-            result = name + acv.getCredentialCrn() + cloudContext.getId() + cloudContext.getAccountId();
+            result = name + acv.getCredentialCrn() + cloudContext.getId() + cloudContext.getTenantId();
         }
```

Comment:
I wonder if this fallback makes any sense at all. (See e.g. [here](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftstorage).)
* The CRN contains colons `:` that are invalid in storage account names.
* Appending the tenant ID right after the (stack) ID without any separators might cause confusions & name clashes.
* Is it possible to lack MD5?

Reply:
- the accountId is not a CRN
- I did not dig in why the result was concatenated this way but I would not change the logic here
- MD5? I don't know. Theoretically not possible that this is missing.